### PR TITLE
[NCL-6954] Add Build Start Button

### PR DESCRIPTION
**Pull request by DnsZhou**

### CODE REVIEW

File: src/components/BuildStartButton/BuildStartButton.tsx [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+            <>
+              <div className={styles['dropdown-icon-item']}>
+                <Checkbox
+                  label="keep pod alive"
+                  isChecked={keepPodOnFailure}
+                  onChange={() => setKeepPodOnFailure(!keepPodOnFailure)}
+                  id="keepPodOnFailure-check"
+                  name="keepPodOnFailure-check"
+                />
+                <span className={'pnc-info-icon'}>
```
Comment (matedo1) [inline src/components/BuildStartButton/BuildStartButton.tsx:310]: why is not using just `"pnc-info-icon"` enough? (the same for other cases)

File: src/components/BuildStartButton/BuildStartButton.tsx [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+      <b>Force</b>
+      <List>
+        <ListItem>Always</ListItem>
+      </List>
+    </div>
+  </>
+);
+
+export const BuildStartButton = ({ buildConfig, groupConfig, size = 'md' }: IBuildStartButtonProp) => {
+  const [isTempBuild, setIsTempBuild] = useState(false);
```
Comment (matedo1) [inline src/components/BuildStartButton/BuildStartButton.tsx:136]: add missing types if possible, for example: `useState<boolean>(false)` (the same for all other cases)

**Review by matedo1 — CHANGES_REQUESTED** [review]
Thanks, just minor changes

File: src/components/BuildStartButton/BuildStartButton.tsx [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+    }
+  };
+
+  return (
+    <>
+      <Button icon={<BuildIcon />} variant="primary" isSmall={size === 'sm'} isLarge={size === 'lg'} onClick={triggerBuild}>
+        Build
+      </Button>
+      <Dropdown
+        alignments={{ sm: 'right', md: 'right', lg: 'right' }}
```
[the remark this answers is not in the document]
Comment (DnsZhou) [reply src/components/BuildStartButton/BuildStartButton.tsx:187]: https://github.com/project-ncl/pnc-web-ui-react/blob/main/src/components/ProjectCreateEditPage/ProjectCreateEditPage.tsx#L31

File: src/components/BuildStartButton/BuildStartButton.module.css [hunk 4]
```diff
@@ -0,0 +1,14 @@
+.dropdown-panel {
+  min-width: 220px;
+  padding: 5px 15px;
+  display: block;
```
[the remark this answers is not in the document]
Comment (DnsZhou) [reply src/components/BuildStartButton/BuildStartButton.module.css:4]: will remove it

**Comment by DnsZhou** [comment]
updated and ready for review @matedo1

**Review by matedo1 — APPROVED** [review]

**Pull request merged**